tests/rbd/test_rbd_trash_purge_schedule.py

```diff
@@ -0,0 +1,152 @@
+            return 1
+        prepare_images_and_move_to_trash(rbd, pool, client, num_images)
+        iterations_remaining -= 1
+
+    log.info(
+        f"Verified trash purge schedule of {schedule_interval} minutes for {verification_iterations} iterations"
+    )
+    return 0
+
+
+def rbd_trash_purge_schedule(rbd, pool_type, **kw):
+    """
+    Orchestrate rbd trash purge schedule test
+    Args:
+        rbd: rbd object
+        pool_type: pool type (ec_pool_config and/or rep_pool_config)
+        **kw: Test data
+    """
+    pool = kw["config"][pool_type]["pool"]
+    num_images = random.randint(2, 5)
+    schedule_interval = random.randint(1, 5)
+    verification_iterations = random.randint(2, 5)
+
+    log.info(
+        f"Trash purge schedule will be tested for an interval of \
+        {schedule_interval} for {verification_iterations} iterations"
+    )
+    client = kw["ceph_cluster"].get_nodes(role="client")[0]
+
+    try:
+        prepare_images_and_move_to_trash(rbd, pool, client, num_images)
+
+        log.info(f"Adding trash purge schedule on pool {pool}")
+        if rbd.trash_purge_schedule(pool, "add", schedule_interval):
+            return 1
+        return verify_schedule_add(
+            rbd, verification_iterations, schedule_interval, pool, client, num_images
+        )
+
+    except Exception as error:
+        log.error(error.message)
+        return 1
+
+    finally:
+        rbd.clean_up(pools=[kw["config"][pool_type]["pool"]])
+
+
+def run(**kw):
+    """Verify the trash purge schedule functionality.
+
+    Args:
+        kw: test data
+
+    Returns:
+        int: The return value. 0 for success, 1 otherwise
+
+    Pre-requisites :
+    We need atleast one client node with ceph-common package,
+    conf and keyring files
+
+    Test cases covered -
+    CEPH-83573660 - Schedule periodic trash purge operations via the CLI and check the behavior
+
+    Test case work flow:
+
+    1. Create some images with i/o
+    2. Move images to trash
+    3. add trash purge schedule
+    4. Verify trash purge schedule is intact and
+    5. images in trash are purged (removed periodically)
+    6. Repeat the above steps for EC-pool.
+    """
+    log.info(
+        "CEPH-83574482 - Running Trash purge test on images with and without clones"
+    )
+    rbd_obj = initial_rbd_config(**kw)
+    if rbd_obj:
+        if "rbd_reppool" in rbd_obj:
+            log.info("Executing test on Replication pool")
+            if rbd_trash_purge_schedule(
+                rbd_obj.get("rbd_reppool"), "rep_pool_config", **kw
+            ):
+                log.error("Verification of schedule on replication pool images failed.")
+                return 1
+        if "rbd_ecpool" in rbd_obj:
+            log.info("Executing test on EC pool")
+            if rbd_trash_purge_schedule(
+                rbd_obj.get("rbd_ecpool"), "ec_pool_config", **kw
+            ):
+                log.error("Verification of schedule on ec pool images failed.")
```

Comment:
can be modified as "Verification is trash purge schedule on ......."

Reply:
Adopted the gist 👍